Quick Start: Update snackbars style on non my site screens

ashiagr opened the pull request:
Fixes #15462

This PR 
- Displays `Quick Start` prompt on other screens (`Page List`, `Reader` and `Publicise`) using the snackbar sequencer. enabling us to remove the functions to [show and hide the `Quick Start` snackbar](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L1565-L1578) from the `WPMainActivity`.
- Updates non-themed style on these snackbars using the [themed style used for `My Site - Quick Start`](https://github.com/wordpress-mobile/WordPress-Android/blob/92dfe6dcb9b6c3f8bd886f95052a455ff0ebfd68/WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteFragment.kt#L230).

To test:

1. Let `Quick Start` card display after creating a new site.
2. Perform below `Quick Start` steps (involving 2-steps, with the second snackbar displaying on the non `My Site` screen )
    Quick Start Step| Target Screen 
    -----------------|---------------
    `Customise your site -> Edit your homepage ` | `Page List`
    `Grow your audience -> Social sharing` | `Publicise`
    `Grow your audience -> Follow other sites`| `Reader`
3. Notice that the snackbar is displayed using the themed style used for prompts on the `My Site` screen.

#### Screenshots for `Quick Start` snackbar on `Page List`
(Same style is applied on the other target screens)

**Light Mode**

Before | After
-------|-------
<img width="442" alt="orig_light" src="https://user-images.githubusercontent.com/1405144/143565538-d3c773af-09d4-4b96-ac6e-99b3bc3e8901.png"> | <img width="441" alt="new_light" src="https://user-images.githubusercontent.com/1405144/143565130-568d5e9c-e9f3-4049-81b4-f4ce83278b8f.png">


**Dark Mode**

Before | After
-------|-------
<img width="441" alt="orig_dark" src="https://user-images.githubusercontent.com/1405144/143565178-6f937732-200b-4ea7-99da-db2d468e9d11.png"> | <img width="439" alt="new_dark" src="https://user-images.githubusercontent.com/1405144/143565196-d2b5d1fe-3f84-44c8-930a-473cd1d8f2f3.png">

**Merge Instructions**
 
1. Make sure design review is done.
2. Remove `Needs Design Review `, `Not Ready for Merge` labels.
3. Merge the PR.

The PR targets `18.9` instead of the `18.8` milestone (even though it can be merged with `18.8`) to avoid any rush for the review.

## Regression Notes
1. Potential unintended areas of impact
N/A (minimal changes done only to update the style).

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ashiagr committed:
Show QS snackbar notice on Reader screen using SnackbarSequencer

ashiagr committed:
Show QS snackbar notice on PublicizeListFragment using SnackbarSequencer

ashiagr committed:
Show QS snackbar notice on PageListFragment using SnackbarSequencer

ashiagr committed:
Remove WPDialogSnackbar, hideQuickStartSnackBar from WPMainActivity

ashiagr committed:
Analysis: Handle detekt warnings

ashiagr committed:
Switch to themed quick start prompt style

ashiagr committed:
Remove unused non-themed quick start prompt style

ashiagr committed:
Analysis: Remove unused dimen

osullivanchris reviewed (approved):
Looks good! The snackbars on non my site screens now look consistent with the ones on my site. Light snackbars appearing in dark mode.

ParaskP7 committed:
Cleanup: Remove unnecessary resource provider from publicize list

ParaskP7 committed:
Refactor: Remove is themed snackbar from stylize quick start prompt

This 'isThemedSnackbar' logic is no longer necessary as every usage of
the 'stylizeQuickStartPrompt(...)' method defaults to 'true'. As such,
even the wrapper name can end up being less complex and as simple as
'stylizeQuickStartPrompt(...)', without the 'Themed' keyword in it.

ParaskP7 committed:
Merge branch 'develop' of github.com:wordpress-mobile/WordPress-Android into issue/15462-update-qs-snackbars-style-on-non-my-site-screens

ParaskP7 reviewed (approved):
👋 @ashiagr !

I have reviewed and tested this PR as per the instructions, everything works as expected, awesome job! 🌟 🌟 🌟 

In addition to your commits:
- I added a couple more commits, nothing major, just a `Cleanup` and a small `Refactoring` to reduce complexity.
- I merged the latest `develop` to resolve the `Lint` issue with the value out of range due to `getColumnIndex(...)`. The newest version is utilizing the `getColumnIndexOrThrow(...)`, which effectively solves/suppresses this warning.

PS: Since @osullivanchris also approved it, I will be merging this PR as soon as CI completes successfully.
